util/src/network/host.rs

```diff
@@ -946,13 +947,17 @@ fn save_key(path: &Path, key: &Secret) {
 		return;
 	};
 	path_buf.push("key");
-	let mut file = match fs::File::create(path_buf.as_path()) {
+	let path = path_buf.as_path();
+	let mut file = match fs::File::create(&path) {
 		Ok(file) => file,
 		Err(e) => {
 			warn!("Error creating key file: {:?}", e);
 			return;
 		}
 	};
+	if let Err(e) = restrict_permissions_owner(&path) {
+		warn!(target: "network", "Failed to modify permissions of the file (chmod: {})", e);
+	}
```

Comment:
might be better to die here rather than continue on in a potentially vulnerable state?